File: remove-dark-classes.js
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

// Get all files with dark: classes
const output = execSync(
  'cd "e:\\EDGEBIC_Web_Description" && grep -r "dark:" "app/(app)/(marketing)" -l',
  { encoding: 'utf-8', shell: 'bash' }
);

const files = output.trim().split('\n').filter(f => f);

console.log(`Found ${files.length} files with dark: classes`);

let processed = 0;
let errors = 0;

files.forEach((file) => {
  try {
    const fullPath = path.join('e:\\EDGEBIC_Web_Description', file);
    let content = fs.readFileSync(fullPath, 'utf-8');

    // Remove dark: prefixed classes using regex
    // This regex matches dark: followed by any non-whitespace characters (the class name)
    // It handles cases like: dark:bg-slate-900, dark:text-white, dark:hover:bg-gray-700
    const originalContent = content;

    // Match dark: classes with various patterns
    content = content.replace(/\s+dark:[^\s"'`}>\]]+/g, '');

    if (content !== originalContent) {
      fs.writeFileSync(fullPath, content, 'utf-8');
      processed++;
      console.log(`✓ Processed: ${file}`);
    } else {
      console.log(`○ No changes: ${file}`);
    }
  } catch (error) {
    errors++;
    console.error(`✗ Error processing ${file}:`, error.message);
  }
});

console.log(`\nSummary:`);
console.log(`- Processed: ${processed} files`);
console.log(`- Errors: ${errors} files`);
console.log(`- Total: ${files.length} files`);

File: remove_dark_classes.py
import os
import re
import subprocess

# Get the marketing directory path
marketing_dir = r"e:\EDGEBIC_Web_Description\app\(app)\(marketing)"

# Get all files with dark: classes using grep
try:
    result = subprocess.run(
        ['grep', '-r', 'dark:', marketing_dir, '-l'],
        capture_output=True,
        text=True,
        shell=False
    )

    files = [f.strip() for f in result.stdout.strip().split('\n') if f.strip()]
    print(f"Found {len(files)} files with dark: classes\n")

    processed = 0
    skipped = 0
    errors = 0

    for file_path in files:
        try:
            # Read the file
            with open(file_path, 'r', encoding='utf-8') as f:
                content = f.read()

            original_content = content

            # Remove dark: classes using regex
            # This pattern matches: space + dark: + any non-whitespace characters
            content = re.sub(r'\s+dark:[^\s"\'`}>\]]+', '', content)

            # Write back if changes were made
            if content != original_content:
                with open(file_path, 'w', encoding='utf-8') as f:
                    f.write(content)
                processed += 1
                print(f"✓ Processed: {file_path}")
            else:
                skipped += 1
                print(f"○ No changes needed: {file_path}")

        except Exception as e:
            errors += 1
            print(f"✗ Error processing {file_path}: {str(e)}")

    print(f"\n{'='*60}")
    print(f"Summary:")
    print(f"  Processed: {processed} files")
    print(f"  Skipped: {skipped} files")
    print(f"  Errors: {errors} files")
    print(f"  Total: {len(files)} files")
    print(f"{'='*60}")

except Exception as e:
    print(f"Error: {str(e)}")
